board/safety/safety_nissan.h

```diff
@@ -0,0 +1,159 @@
+
+
+const int NISSAN_MAX_STEER = 2047; // 1s
+// real time torque limit to prevent controls spamming
+// the real time limit is 1500/sec
+const int NISSAN_MAX_RT_DELTA = 940;          // max delta torque allowed for real time checks
+const uint32_t NISSAN_RT_INTERVAL = 250000;    // 250ms between real time checks
+const int NISSAN_MAX_RATE_UP = 50;
+const int NISSAN_MAX_RATE_DOWN = 70;
+const int NISSAN_DRIVER_TORQUE_ALLOWANCE = 60;
+const int NISSAN_DRIVER_TORQUE_FACTOR = 10;
+
+float nissan_speed = 0;
+int nissan_controls_allowed_last = 0;
+int nissan_angle_meas_new = 0;
+uint32_t nissan_ts_angle_last = 0;
+int nissan_cruise_engaged_last = 0;
+int nissan_rt_angle_last = 0;
+int nissan_desired_angle_last = 0;
+uint32_t nissan_ts_last = 0;
+struct sample_t nissan_torque_driver; // last few driver torques measured
+
+
+const struct lookup_t NISSAN_LOOKUP_ANGLE_RATE_UP = {
+  {2., 7., 17.},
+  {5., .8, .15}};
+
+const struct lookup_t NISSAN_LOOKUP_ANGLE_RATE_DOWN = {
+  {2., 7., 17.},
+  {5., 3.5, .4}};
+
+// TODO: Need to work out the max angle at certain speeds
+const struct lookup_t NISSAN_LOOKUP_MAX_ANGLE = {
+    {2., 29., 38.},
+    {500., 500., 500.}};
+
+
+static void nissan_rx_hook(CAN_FIFOMailBox_TypeDef *to_push) {
+  int bus = GET_BUS(to_push);
+  int addr = GET_ADDR(to_push);
+
+  if (bus == 0) {
+    if (addr == 0x2) {
+      // Current steering angle
+      // Factor -0.1, little endian
+      nissan_angle_meas_new = to_signed(to_push->RDLR & 0xFFFF, 16) * -0.1;
```

Comment:
we now have the convenient macros `GET_BYTE`. Would you mind using that instead?